### PR TITLE
CONSERVE statements processing (issues #280, #284)

### PR DESCRIPTION
**Pull request by cattabiani**
- Bugfix (issue #284): conserve statements in derivative blocks are not corrupt anymore
- Bugfix (issue #280): KineticBlockVisitor visits only kinetic blocks (no more seg. fault)
- Added a warning to enlight the user that the nmodl printed in between the KineticBlockVisitor and the SympyBlockVisitor, in presence of CONSERVE statements, is not standard (for example it does not work in mod2c). It should be used only for debugging/developing

### CODE REVIEW

File: src/nmodl/main.cpp [hunk 1]
```diff
@@ -356,7 +356,13 @@ int main(int argc, const char* argv[]) {
...
             KineticBlockVisitor().visit_program(ast.get());
             SymtabVisitor(update_symtab).visit_program(ast.get());
-            ast_to_nmodl(ast.get(), filepath("kinetic"));
+            const auto filename = filepath("kinetic");
+            ast_to_nmodl(ast.get(), filename);
+            if (nmodl_ast) {
+                logger->warn(
+                    "{} may present non-standard CONSERVE statements in DERIVATIVE blocks. In this case, use it only for debugging/developing"_format(
+                        filename));
+            }
```
Comment (pramodk) [inline src/nmodl/main.cpp:365]: * Here we will always generate this warning even if CONSERVE statement doesn't exist.
* I would prefer to do lookup in the AST and see if there is any CONSERVE statement inside DERIVATIVE block and then only issue warning. Its extra lookup cost but would be cleaner.

File: src/visitors/kinetic_block_visitor.cpp [hunk 2]
```diff
@@ -459,8 +464,13 @@ void KineticBlockVisitor::visit_program(ast::Program* node) {
...
     }
 
+    auto kineticBlockNodes = AstLookupVisitor().lookup(node, ast::AstNodeType::KINETIC_BLOCK);
     // replace reaction statements within each kinetic block with equivalent ODEs
-    node->visit_children(*this);
+    for (const auto& ii: kineticBlockNodes) {
+        ii->accept(*this);
+    }
+
+    //    node->visit_children(*this);
```
Comment (pramodk) [inline src/visitors/kinetic_block_visitor.cpp:473]:
```suggestion
```

File: src/nmodl/main.cpp (hunk 1 above)
Comment (cattabiani) [reply src/nmodl/main.cpp:365]: I think I have an idea to not do another lookup...

Comment (cattabiani) [reply src/nmodl/main.cpp:365]: done!

File: src/visitors/kinetic_block_visitor.hpp [hunk 3]
```diff
@@ -132,6 +135,9 @@ class KineticBlockVisitor: public AstVisitor {
 
   public:
     KineticBlockVisitor() = default;
```
Comment (pramodk) [inline src/visitors/kinetic_block_visitor.hpp:137]:
```suggestion
    KineticBlockVisitor() = default;
    
```

File: src/visitors/kinetic_block_visitor.hpp [hunk 4]
```diff
@@ -132,6 +135,9 @@ class KineticBlockVisitor: public AstVisitor {
...
+    inline int get_conserve_statement_count() const {
```
Comment (pramodk) [inline src/visitors/kinetic_block_visitor.hpp:138]: nitpick: 
```suggestion
    int get_conserve_statement_count() const {
```

File: src/visitors/kinetic_block_visitor.cpp [hunk 5]
```diff
@@ -459,8 +466,11 @@ void KineticBlockVisitor::visit_program(ast::Program* node) {
         }
     }
 
+    auto kineticBlockNodes = AstLookupVisitor().lookup(node, ast::AstNodeType::KINETIC_BLOCK);
     // replace reaction statements within each kinetic block with equivalent ODEs
-    node->visit_children(*this);
+    for (const auto& ii: kineticBlockNodes) {
```
Comment (pramodk) [inline src/visitors/kinetic_block_visitor.cpp:471]:
```suggestion
    for (const auto& ii: kinetic_block_nodes) {
```

**Pull request merged**